### ISSUE TITLE
Add Documentation for Output Tensors

### ISSUE THREAD

**Issue by J-Britten** [ultralytics/ultralytics#16944]
When converting a YOLO model to ONNX (or other), we are told the model's output shape(s). 

Since I can't use the ultralytics package to rely on the automatically created results object for my use case, I have to generate the output (e.g. masks) myself using the output tensors.

However, no documentation is provided on what these tensors represent. 

After some research, I found [this issue](https://github.com/ultralytics/ultralytics/issues/16238) and was able to use this as a starting point.

It would be nice to have this information readily available in the documentation for those who need it instead of relying on old issues.

---
If you came looking for information on how these output tensors work: I used YOLO11n-seg, converted it to ONNX following the [documentation](https://docs.ultralytics.com/tasks/segment/#how-can-i-export-a-yolo-segmentation-model-to-onnx-format) with no additional parameters:

The output tensor __(1, 116, 8400)__ consists of 

**batch size x (4 bounding box values + 80 classes from COCO dataset + 32 mask coefficients) x number of detected objects**

YOLO detects up to 8400 objects per image, you need to filter them using NMS

The bounding box values are: **centerX, centerY, width, height**

The 32 mask coefficients are needed to create the mask in combination with the second output tensor:
The output tensor **(1, 32, 160, 160)** consists of **batch size x mask layers x width x height**

To get a mask of an object, you need to multiply its mask coefficients with their respective mask layers, then add up all the mask layers. 
You will notice that there are still some other elements on the mask. To clean that up, you can use the bounding box

**Comment by UltralyticsAssistant** [comment]
👋 Hello @J-Britten, thank you for your input on the Ultralytics repository 🚀! We're glad you're diving deep into model conversion and understanding output tensors. To enhance our documentation, your observations are invaluable.

For users facing similar challenges, we recommend checking the [Documentation](https://docs.ultralytics.com) for guidance on tasks like model export. However, we understand specific details like tensor output shapes might not be fully covered yet.

If this is a 🐛 Bug Report regarding the existing documentation or tools, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us tackle the issue efficiently.

Interested in custom insights or questions ❓? Please share any dataset examples and logs you are using. Ensuring you're aligned with our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) is always helpful.

Join discussions in the Ultralytics community that suit your preferences. For immediate conversations, jump into our [Discord](https://ultralytics.com/discord) 🎧. Prefer comprehensive discussions? [Discourse](https://community.ultralytics.com) is ideal for that. You can even explore trending discussions on our [Subreddit](https://reddit.com/r/ultralytics) 🌟.

## Upgrade

Ensure you're using the latest version of `ultralytics`, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) under a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Run YOLO in these verified environments with dependencies such as [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn) pre-installed:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, the [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) verifies the functionality of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across various operating systems daily.

Note that this is an automated response aimed at guiding you to useful resources. An Ultralytics engineer will follow up to assist further. Thank you for contributing and helping us improve our tools and documentation! 🌟

**Comment by Y-T-G** [comment]
There are also examples showing how to process the output.

https://github.com/ultralytics/ultralytics/tree/main/examples
